client/packages/host/src/Admin/Settings.tsx

```diff
@@ -179,6 +181,10 @@ export const Settings: React.FC = () => {
       </>
     ) : null;
 
+  useEffect(() => {
```

Comment:
this should be in `useNativeClient` but having the internal state in there wasn't working. The hook wasn't refreshing the value in the component that was using it. Since there's only two places that its used (here and `Android.tsx`), I have moved it out to those.

Reply:
I think this would be the only real place that we would need to use `preferences` since url may not match discovery url, at this point I would just add url parameter to discovery page (that would trigger overwrite the localStorage)